rspace/src/main/scala/coop/rchain/rspace/RSpace.scala

```diff
@@ -110,17 +110,20 @@ class RSpace[C, P, A, K](val store: IStore[C, P, A, K]) extends ISpace[C, P, A,
                   |at <channels: $channels>""".stripMargin.replace('\n', ' '))
             None
           case Some(dataCandidates) =>
-            dataCandidates
-              .sortBy(_.datumIndex)(Ordering[Int].reverse)
-              .foreach {
-                case DataCandidate(candidateChannel, Datum(_, persistData), dataIndex)
-                    if !persistData =>
-                  store.removeDatum(txn, Seq(candidateChannel), dataIndex)
-                case _ =>
-                  ()
-              }
-            logger.debug(s"consume: data found for <patterns: $patterns> at <channels: $channels>")
-            Some((continuation, dataCandidates.map(_.datum.a)))
+            store.eventsCounter.registerConsumeCommEvent {
```

Comment:
You are still measuring the duration of this block - **do not do this**.  This block _is not_ the COMM Event, so the duration of this block is meaningless - it is not the duration of a COMM Event.  **COMM Events do not have a duration**.  

As I said before, you should be only be incrementing the counter and the throughput statistics on this line.